README: add transparency on affiliations (#119)
NixOS/foundation: README.md

Board observers are here to help and participate in the board decision making
process, but don't have any voting powers.

### Board affiliation transparency

> * Eelco Dolstra is Founder / Secretary in the Board of Determinate Systems

Reviewer: ```suggestion
* Eelco Dolstra is Founder / Secretary of the Board of Determinate Systems
```

Author: Thanks, done!

> * Eelco Dolstra is [a secretary of Determinate Systems, Inc.](https://corp.sec.state.ma.us/CorpWeb/CorpSearch/CorpSummary.aspx?sysvalue=OHpcYT1dK2Z48whEuewVuhiIp5yy7VpR1UIbx7EnDno-)

Reviewer 2: ```suggestion
* Eelco Dolstra is [Founder of Determinate Systems, Inc.](https://corp.sec.state.ma.us/CorpWeb/CorpSearch/CorpSummary.aspx?sysvalue=OHpcYT1dK2Z48whEuewVuhiIp5yy7VpR1UIbx7EnDno-)
```

Author: @grahamc Do you know the role of Eelco w.r.t. to the board? I will put founder in addition but it could be that Eelco or anyone in those list leave their role and become only founders / creators / etc. I think it makes sense for this document to list "active state".

Reviewer: "Secretary" (not "a secretary") or just "board member".

Reviewer: And drop the link, it's not really relevant.

Author: done!

* Ron Efroni is Chief Executive Officer of Flox

> * Théophane Hufschmitt is employed by Tweag/Modus

Reviewer 3: ```suggestion
* Théophane Hufschmitt is employed by Tweag/Modus Create
```

Author: Addressed.

> * Domen Kozar is Chief Executive Officer of Tweag

Reviewer 4: ```suggestion
* Domen Kozar is Chief Executive Officer of Cachix
```

Author: brain so stupid

> * Domen Kozar is Chief Executive Officer of Cachix

Reviewer 5: ```suggestion
* Domen Kozar is Founder/Director of Cachix
```

We don't have hierarchy really at the typical company structure, so this is more appropriate.

Author: Done!
